src/data/index.ts

```diff
@@ -126,7 +194,11 @@ export class Index<T extends RecordMetadata = RecordMetadata> {
    * @returns A promise that resolves when the delete is completed.
    * @throws {@link Errors.PineconeArgumentError} when invalid arguments are passed.
    */
-  deleteOne: ReturnType<typeof deleteOne>;
+  deleteOne(id) {
+    return this._deleteOne(id);
+  }
+  /** @hidden */
+  _deleteOne: ReturnType<typeof deleteOne>;
```

Comment:
This and all similar changes to `Index` and `Pinecone` are about helping typedoc display method information in a consistent way. If you assign a function to a property in the constructor, it treats it as a property instead of as a method and having inconsistency there made the docs look ugly/confusing.